virtual_rainforest/models/animals/dummy_animal_module.py

```diff
@@ -78,6 +82,75 @@ class PalatableSoil:
     """The grid position of the soil pool."""
 
 
+class AnimalCommunity:
+    """This is a class for the animal community of a grid cell."""
+
+    def __init__(
+        self,
+        functional_groups: list[FunctionalGroup],
+    ) -> None:
+        """The constructor of the AnimalCommunity class."""
+        self.functional_groups = functional_groups
+        """A list of all FunctionalGroup types in the model."""
+
+        self.cohorts: dict[str, list[AnimalCohort]] = {
+            k.name: [] for k in self.functional_groups
+        }
+        """Generate a dictionary of functional groups."""
+
+    def immigrate(self, cohort: AnimalCohort, destination: AnimalCommunity) -> None:
+        """Function to move an AnimalCohort between AnimalCommunity objects.
+
+        This function should take a cohort and a destination community and then pop the
+        cohort from this community to the destination.
+
+        Args:
+            cohort: The AnimalCohort moving between AnimalCommunities.
+            destination: The AnimalCommunity the cohort is moving to.
+
+        """
+        immigrant = self.cohorts[cohort.name].pop(
+            self.cohorts[cohort.name].index(cohort)
+        )
+        # Remove target cohort from source community
+        destination.cohorts[immigrant.name].append(immigrant)
+        # Add cohort to destination community
+
+    def die_cohort(self, cohort: AnimalCohort) -> None:
+        """The function to change the cohort status from alive to dead.
+
+        Args:
+            cohort: The AnimalCohort instance that has lost all individuals.
+
+        """
+
+        if cohort.is_alive:
+            cohort.is_alive = False
+            LOGGER.info("An animal cohort has died")
+            self.cohorts[cohort.name].remove(cohort)
```

Comment:
I'd consider making this a `DEBUG` log entry? We're going to need to do a systematic review of logging entries at some point, but this seems like `DEBUG`. It would be good to report more specific data too - cohort type and maybe even grid id if available?

I think there's some redundancy here too - `self.cohorts` is a list of alive cohorts and if there isn't a conceivable mechanism where dead cohort could be there, then setting and testing flags might be superfluous? For example, setting `is.alive = False` here is immediately followed by removing and discarding that object.